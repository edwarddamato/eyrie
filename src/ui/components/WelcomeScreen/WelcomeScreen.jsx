import React from 'react';
import PropTypes from 'prop-types';
import CreateVault from '../CreateVault';
import './WelcomeScreen.scss';

const ACTIVE_CHOICE = {
  NONE: 0,
  CREATE: 1,
  OPEN: 2
};

class WelcomeScreen extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      activeChoice: ACTIVE_CHOICE.NONE
    };

    this.handleCreateVault = this.handleCreateVault.bind(this);
    this.handleOpenVault = this.handleOpenVault.bind(this);
  }

  handleCreateVault () {
    this.setState({
      activeChoice: ACTIVE_CHOICE.CREATE
    });
  }

  handleOpenVault () {
    this.setState({
      activeChoice: ACTIVE_CHOICE.OPEN
    });
  }

  render () {
    switch (this.state.activeChoice) {
      default:
      case ACTIVE_CHOICE.NONE:
        return (
          <section className="ws">
            <h2 className="ws_title">{'Let\'s start.'}</h2>
            <ul className="ws_choice-list">
              <li className="ws_choice-item"><button className="ws_choice-button" onClick={this.handleCreateVault}>Create new vault</button></li>
              <li className="ws_choice-item"><button className="ws_choice-button" onClick={this.handleOpenVault}>Open existing vault</button></li>
            </ul>
          </section>
        );
      case ACTIVE_CHOICE.CREATE:
        return <CreateVault loadingHandler={this.props.loadingHandler} />;
      case ACTIVE_CHOICE.OPEN:
        return <div>OPEN</div>;
    }
  }
}
WelcomeScreen.propTypes = {
  loadingHandler: PropTypes.func.isRequired
};

export default WelcomeScreen;
