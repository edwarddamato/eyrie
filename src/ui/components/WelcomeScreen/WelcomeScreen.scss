.ws {
  width: 45%;

  &_title {
  }

  &_choice {
    &-list {

    }
    &-item {
    }
    &-button {
      margin: 1rem 0;
      width: 100%;
    }
  }
}