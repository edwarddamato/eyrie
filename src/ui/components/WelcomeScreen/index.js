import WelcomeScreen from './WelcomeScreen.jsx';
export default WelcomeScreen;
