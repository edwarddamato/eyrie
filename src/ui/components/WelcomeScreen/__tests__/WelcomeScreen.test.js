import React from 'react';
import renderer from 'react-test-renderer';
import WelcomeScreen from '../';

jest.mock('electron', () => {
  return {
    remote: {
      require: () => {
        return {
          dialog: {

          }
        };
      }
    }
  };
});

describe('WelcomeScreen', () => {
  it('renders correctly', () => {
    const welcomeScreen = renderer.create(
      <WelcomeScreen loadingHandler={() => { }} />
    ).toJSON();
    expect(welcomeScreen).toMatchSnapshot();
  });
});
