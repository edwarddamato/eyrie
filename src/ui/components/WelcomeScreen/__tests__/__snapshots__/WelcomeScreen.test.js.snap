// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen renders correctly 1`] = `
<section
  className="ws"
>
  <h2
    className="ws_title"
  >
    Let's start.
  </h2>
  <ul
    className="ws_choice-list"
  >
    <li
      className="ws_choice-item"
    >
      <button
        className="ws_choice-button"
        onClick={[Function]}
      >
        Create new vault
      </button>
    </li>
    <li
      className="ws_choice-item"
    >
      <button
        className="ws_choice-button"
        onClick={[Function]}
      >
        Open existing vault
      </button>
    </li>
  </ul>
</section>
`;
